loki/frontend/fparser.py

```diff
@@ -2483,6 +2483,42 @@ def visit_Call_Stmt(self, o, **kwargs):
             arguments = tuple(arg for arg in arguments if not isinstance(arg, tuple))
         else:
             arguments, kwarguments = (), ()
+        # Figure out the exact procedure being called if this is a call to a generic interface
+        if name.type.dtype.is_generic:
+            # If the interface is imported, take its definition from a module
+            # TODO: handle interfaces defined in the same module
+            if name.type.imported:
+                module = name.type.module
+                interface = [i for i in module.interfaces if i.spec.name == name.name]
+                if len(interface) == 1:
+                    interface = interface[0]
```

Comment:
You should be able to shortcut this with
```suggestion
                interface = module.interface_map[name.name]
```